SentinelOne/cloud_funnel2.0/ingest/parser.yml

```
@@ -0,0 +1,661 @@
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_source_process_start
+    filter: "{{json_event.message.event.source != null and json_event.message.event.source.fullPid.startTime.millisecondsSinceEpoch != null and json_event.message.event.source.fullPid.startTime.millisecondsSinceEpoch | length == 13}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.source.fullPid.startTime.millisecondsSinceEpoch[:-3]}}.{{json_event.message.event.source.fullPid.startTime.millisecondsSinceEpoch[-3:]}}"
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_process_start
+    filter: "{{json_event.message.event.process != null and json_event.message.event.process.fullPid.startTime.millisecondsSinceEpoch != null and json_event.message.event.process.fullPid.startTime.millisecondsSinceEpoch | length == 13}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.process.fullPid.startTime.millisecondsSinceEpoch[:-3]}}.{{json_event.message.event.process.fullPid.startTime.millisecondsSinceEpoch[-3:]}}"
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_parent_process_start
+    filter: "{{json_event.message.event.parent != null and json_event.message.event.parent.fullPid.startTime.millisecondsSinceEpoch != null and json_event.message.event.parent.fullPid.startTime.millisecondsSinceEpoch | length == 13}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.parent.fullPid.startTime.millisecondsSinceEpoch[:-3]}}.{{json_event.message.event.parent.fullPid.startTime.millisecondsSinceEpoch[-3:]}}"
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_target_process_start
+    filter: "{{json_event.message.event.target != null and json_event.message.event.target.fullPid.startTime.millisecondsSinceEpoch != null and json_event.message.event.target.fullPid.startTime.millisecondsSinceEpoch | length == 13}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.target.fullPid.startTime.millisecondsSinceEpoch[:-3]}}.{{json_event.message.event.target.fullPid.startTime.millisecondsSinceEpoch[-3:]}}"
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_file_creation_time
+    filter: "{{json_event.message.event.file != null and json_event.message.event.file.creationTime.millisecondsSinceEpoch != null and json_event.message.event.file.creationTime.millisecondsSinceEpoch | length == 13}}"
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.file.creationTime.millisecondsSinceEpoch[:-3]}}.{{json_event.message.event.file.creationTime.millisecondsSinceEpoch[-3:]}}"
```

`date.parse` now supports milliseconds timestamp
```suggestion
    filter: "{{json_event.message.event.file != null and json_event.message.event.file.creationTime.millisecondsSinceEpoch != null}}"
    external:
      name: date.parse
      properties:
        input_field: "{{json_event.message.event.file.creationTime.millisecondsSinceEpoch}}"
```